src/GameLogic/AttackableExtensions.cs

```diff
@@ -34,13 +34,14 @@ public static class AttackableExtensions
     /// <param name="isCombo">If set to <c>true</c>, the damage gets increased by a combo bonus.</param>
     /// <param name="damageFactor">The damage factor.</param>
     /// <returns>
-    /// The hit information.
+    /// The hit information, and the soul barrier mana toll, if any.
     /// </returns>
-    public static async ValueTask<HitInfo> CalculateDamageAsync(this IAttacker attacker, IAttackable defender, SkillEntry? skill, bool isCombo, double damageFactor = 1.0)
+    public static async ValueTask<(HitInfo HitInfo, float ManaToll)> CalculateDamageAsync(this IAttacker attacker, IAttackable defender, SkillEntry? skill, bool isCombo, double damageFactor = 1.0)
     {
+        float manaToll = 0;
         if (!attacker.IsAttackSuccessfulTo(defender))
         {
-            return new HitInfo(0, 0, DamageAttributes.Undefined);
+            return (new HitInfo(0, 0, DamageAttributes.Undefined), manaToll);
```

Comment:
What about adding `ManaToll` or `ManaReduction` optionally to `HitInfo` as well? 😉